internal/registry/registry.go

```diff
@@ -142,13 +135,17 @@ func (r *registry) getPullOptions(ctx context.Context, image string, po *kmmv1be
 	return &RepoPullConfig{repo: repo, authOptions: options}, nil
 }
 
-func (r *registry) getImageManifest(ctx context.Context, image string, pullConfig *RepoPullConfig) ([]byte, error) {
+func (r *registry) getImageManifest(ctx context.Context, image string, po *kmmv1beta1.PullOptions, registryAuthGetter auth.RegistryAuthGetter) ([]byte, *RepoPullConfig, error) {
```

Comment:
I kind of liked the fact that we are detaching the "transport" options from the other parameters. Looks cleaner to me.

Reply:
I probably missed it u/s :( 
I think it complicates it but it is still better to keep the sync with u/s.
We can always change it u/s later.